src/test/java/org/folio/rest/impl/GOBIIntegrationServiceResourceImplTest.java

```diff
@@ -614,6 +621,223 @@ public final void testPostGobiOrdersFallBackLocation(TestContext context) throws
+    logger.info("Begin: Testing for 201 - Return existing Order even if it is pending, and retry to Open succeeds");
+
+    final String body = getMockData(POLISTEDELECTRONICSERIALPATH);
+
+    final GobiResponse order = RestAssured
+      .given()
+        .header(TOKENHEADER)
+        .header(URLHEADER)
+        .header(TENANTHEADER)
+        .header(CONTENTTYPEHEADERXML)
+        .header(new Header(MOCK_OKAPI_GET_ORDER_HEADER, MOCK_INSTRUCTION_GET_PENDING_ORDER))
+        .body(body)
+      .when()
+        .post(ORDERSPATH)
+      .then()
+        .statusCode(201)
+        .contentType(ContentType.XML)
+        .extract()
+          .body()
+            .as(GobiResponse.class, ObjectMapperType.JAXB);
+
+    //should fetch existing order which is in Pending state
+    List<JsonObject> getOrder = MockServer.serverRqRs.get(PURCHASEORDER, HttpMethod.GET);
+    assertEquals(1, getOrder.size());
+
+    //should try to Open, the existing Pending Order, which succeds
+    List<JsonObject> putOrder = MockServer.serverRqRs.get(COMPOSITE_PURCHASE_ORDER, HttpMethod.PUT);
+    assertEquals(1, putOrder.size());
+
+    //should not try to create a new Order
+    List<JsonObject> postedOrder = MockServer.serverRqRs.get(PURCHASEORDER, HttpMethod.POST);
+    assertNull(postedOrder);
+
+    //return the existing PO Line Number
+    assertNotNull(order.getPoLineNumber());
+
+
+    logger.info("End: Testing for 201 - Return existing Order even if it is pending, and retry to Open suceeds");
+  }
+
+  @Test
+  public final void testPostGobiOrdersGetExistingPOLineNumberFails() throws Exception {
+    logger.info("Begin: Testing for 201 - Create new Order , if fetching existing PO Line Number Fails");
+
+    final String body = getMockData(POLISTEDELECTRONICSERIALPATH);
+
+    final GobiResponse order = RestAssured
+      .given()
+        .header(TOKENHEADER)
+        .header(URLHEADER)
+        .header(TENANTHEADER)
+        .header(CONTENTTYPEHEADERXML)
+        .header(new Header(MOCK_OKAPI_GET_ORDER_HEADER, MOCK_INSTRUCTION_GET_OPEN_ORDER))
+        .header(new Header(MOCK_OKAPI_GET_ORDER_BY_ID_HEADER, MOCK_INSTRUCTION_GET_BYID_FAIL))
+        .body(body)
+      .when()
+        .post(ORDERSPATH)
+      .then()
+        .statusCode(201)
+        .contentType(ContentType.XML)
+        .extract()
+          .body()
+            .as(GobiResponse.class, ObjectMapperType.JAXB);
+
+    //should fetch existing order which is in Open state
+    List<JsonObject> getOrder = MockServer.serverRqRs.get(PURCHASEORDER, HttpMethod.GET);
+    assertEquals(1, getOrder.size());
+
+    //should not retry to Open
+    List<JsonObject> putOrder = MockServer.serverRqRs.get(COMPOSITE_PURCHASE_ORDER, HttpMethod.PUT);
+    assertNull(putOrder);
+
+    //should try to create a new Order
+    List<JsonObject> postedOrder = MockServer.serverRqRs.get(PURCHASEORDER, HttpMethod.POST);
+    assertEquals(1,postedOrder.size());
```

Comment:
space missing between `expected` and `actual`

Reply:
Done